core/src/main/java/io/questdb/griffin/SqlUtil.java

```diff
@@ -241,6 +260,74 @@ public static CharSequence fetchNext(GenericLexer lexer) throws SqlException {
         return null;
     }
 
+    /**
+     * Fetches the next non-whitespace, non-comment hint token from the lexer.
+     * <p>
+     * This method should only be called after entering a hint block. Specifically,
+     * a previous call to {@link #fetchNext(GenericLexer, boolean)} must have returned
+     * a hint start token (<code>/*+</code>) before this method can be used.
+     * <p>
+     * The method processes the input stream, skipping over any nested comments and whitespace,
+     * and returns the next meaningful hint token. This allows for clean parsing of hint
+     * content without manual handling of comments and formatting characters.
+     * <p>
+     * When the end of the hint block is reached, the method returns null, indicating
+     * no more hint tokens are available for processing.
+     *
+     * @param lexer The input lexer containing the token stream to process
+     * @return The next meaningful hint token, or null if the end of the hint block is reached
+     * @throws SqlException If a parsing error occurs while processing the hint tokens
+     * @see #fetchNext(GenericLexer, boolean) For entering the hint block initially
+     */
+    public static CharSequence fetchNextHintToken(GenericLexer lexer) throws SqlException {
```

Comment:
if we are being tolerant to syntax errors here, we should not be throwing exception out